diff --git a/package.json b/package.json
index 81ba078baf1445f903eb6a94941b4b52159b8eb9..3bfe2c5714845330c57281be776bb37e1155ab37 100644
--- a/package.json
+++ b/package.json
@@ -72,7 +72,8 @@
     "qrcode": "^1.4.4",
     "safe-decode-uri-component": "^1.2.1",
     "tunnel": "^0.0.6",
-    "yargs": "^17.1.1"
+    "yargs": "^17.1.1",
+    "mri": "^1.2.0"
   },
   "devDependencies": {
     "@types/express": "^4.17.13",
diff --git a/server.js b/server.js
index 4f6602965f40d449157a5d31b3dfc4e10322e4bc..6eb8c30be979468effccad9f4fe7556ecdb2c170 100644
--- a/server.js
+++ b/server.js
@@ -8,7 +8,9 @@ const cache = require('./util/apicache').middleware
 const { cookieToJson } = require('./util/index')
 const fileUpload = require('express-fileupload')
 const decode = require('safe-decode-uri-component')
+const mri = require('mri')
 
+const argv = mri(process.argv.slice(2))
 /**
  * The version check result.
  * @readonly
@@ -141,6 +143,9 @@ async function consturctServer(moduleDefs) {
    */
   app.use((req, res, next) => {
     if (req.path !== '/' && !req.path.includes('.')) {
+      if (argv.proxy) {
+        req.query.proxy = argv.proxy
+      }
       res.set({
         'Access-Control-Allow-Credentials': true,
         'Access-Control-Allow-Origin':
diff --git a/util/request.js b/util/request.js
index cb620670da93b28681cd8cba336234379e0b787d..a5e6588314b25e11890f11772c19d5221467b8b6 100644
--- a/util/request.js
+++ b/util/request.js
@@ -147,10 +147,11 @@ const createRequest = (method, url, data = {}, options) => {
       } else {
         const purl = new URL(options.proxy)
         if (purl.hostname) {
-          const agent = tunnel.httpsOverHttp({
+          const agent = tunnel[purl.protocol === 'https' ? 'httpsOverHttp' : 'httpOverHttp']({
             proxy: {
               host: purl.hostname,
               port: purl.port || 80,
+              proxyAuth: purl.username && purl.password ? (purl.username + ':' + purl.password) : ''
             },
           })
           settings.httpsAgent = agent