diff --git a/CHANGELOG.MD b/CHANGELOG.MD
deleted file mode 100644
index a67472ece42b6b5d032b8a54d285537d7413c979..0000000000000000000000000000000000000000
diff --git a/package.json b/package.json
index cd8608a1af5562ba20c3cfe7c82fdbc639e1e40b..69728f711e00c8511ae998e581a22d9eaae9a927 100644
--- a/package.json
+++ b/package.json
@@ -67,6 +67,7 @@
     "express": "^4.17.1",
     "express-fileupload": "^1.1.9",
     "md5": "^2.3.0",
+    "mri": "^1.2.0",
     "music-metadata": "^7.5.3",
     "pac-proxy-agent": "^7.0.0",
     "qrcode": "^1.4.4",
diff --git a/server.js b/server.js
index befa2866fb91430f82953314a92e43b6ccae33d9..f9a4a96588c816ff417852e0de7840d6bcbee7cc 100644
--- a/server.js
+++ b/server.js
@@ -9,6 +9,7 @@ const { cookieToJson } = require('./util/index')
 const fileUpload = require('express-fileupload')
 const decode = require('safe-decode-uri-component')
 
+const argv = mri(process.argv.slice(2))
 /**
  * The version check result.
  * @readonly
@@ -141,6 +142,9 @@ async function consturctServer(moduleDefs) {
    */
   app.use((req, res, next) => {
     if (req.path !== '/' && !req.path.includes('.')) {
+      if (argv.proxy) {
+        req.query.proxy = argv.proxy
+      }
       res.set({
         'Access-Control-Allow-Credentials': true,
         'Access-Control-Allow-Origin':
diff --git a/util/request.js b/util/request.js
index e30a404c32a5afb0b70346bab5ddd0a73811a387..6bf1628c01f2be3a61563a36ebbadfa8a1b0de10 100644
--- a/util/request.js
+++ b/util/request.js
@@ -147,10 +147,11 @@ const createRequest = (method, url, data = {}, options) => {
       } else {
         const purl = new URL(options.proxy)
         if (purl.hostname) {
-          const agent = tunnel.httpsOverHttp({
+          const agent = tunnel[purl.protocol === 'https' ? 'httpsOverHttp' : 'httpOverHttp']({
             proxy: {
               host: purl.hostname,
               port: purl.port || 80,
+              proxyAuth: purl.username && purl.password ? (purl.username + ':' + purl.password) : ''
             },
           })
           settings.httpsAgent = agent